lib/mongo/error/operation_failure.rb

```diff
@@ -77,12 +94,73 @@ def retryable?
       # @example Is the error retryable for writes?
       #   error.write_retryable?
       #
-      # @return [ true, false ] If the error is retryable.
+      # @return [ true, false ] Whether the error is retryable.
       #
       # @since 2.4.2
       def write_retryable?
-        WRITE_RETRY_MESSAGES.any? { |m| message.include?(m) }
+        WRITE_RETRY_MESSAGES.any? { |m| message.include?(m) } ||
+        write_retryable_code?
+      end
+
+      def write_retryable_code?
+        if code
```

Comment:
I think this could be made more succient by just doing `code && WRITE_RETRY_ERRORS.any? { |e| e[:code] == code }`

Reply:
That would evaluate to nil if code is nil, which it can be.